### PR TITLE
Log and leak

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: libfreerdp/core/credssp_auth.c [hunk 1]
```diff
@@ -104,7 +104,7 @@ rdpCredsspAuth* credssp_auth_new(const rdpContext* rdp_ctx)
 
 BOOL credssp_auth_init(rdpCredsspAuth* auth, TCHAR* pkg_name, SecPkgContext_Bindings* bindings)
 {
-	SECURITY_STATUS status;
+	SECURITY_STATUS status = ERROR_INTERNAL_ERROR;
```
Comment (hardening) [inline libfreerdp/core/credssp_auth.c:107]: This is not needed, `status` is unconditionality set

**Comment by hardening** [comment]
IMHO the changes in libfreerdp/core/credssp_auth.c just had noise in the code by initializing status with values even when it is unconditionally set after. One way to workaround this could be to move the variable declaration at the first affectation.

**Comment by hardening** [comment]
And the same applies to nla.c, over-initializing variables just had noise. Especially when you declare the variable initialized to NULL at the beginning of the function, and then affect it 3 lines later. As we've changed the rules for variables declared in the middle of a block, let's take advantage of that.

File: libfreerdp/core/credssp_auth.c (hunk 1 above)
Comment (akallabeth) [reply libfreerdp/core/credssp_auth.c:107]: will change to declare where used where possible.

**Review by mfleisz — APPROVED** [review]
LGTM

**Pull request merged**